packages/eslint-plugin/docs/rules/call-super-on-override.md

```diff
@@ -0,0 +1,43 @@
+---
+description: 'Require overridden methods to call super.method in their body.'
+---
+
+> 🛑 This file is source code, not the primary documentation location! 🛑
+>
+> See **https://typescript-eslint.io/rules/call-super-on-override** for documentation.
+
+This rule enforces that overridden methods are calling exact super method to avoid missing super class method implementations.
+
+## Rule Details
+
+Examples of code for this rule:
+
+### ❌ Incorrect
+
+```ts
+class Foo1 {
+  bar(param: any): void {}
+}
+
+class Foo2 extends Foo1 {
+  override bar(param: any): void {}
+}
+```
+
+### ✅ Correct
+
+```ts
+class Foo1 {
+  bar(param: any): void {}
+}
+
+class Foo2 extends Foo1 {
+  override bar(param: any): void {
+    super.bar(param);
+  }
+}
+```
+
+## When Not To Use It
+
+When you are using TypeScript < 4.3 or you did not set `noImplicitOverride: true` in `CompilerOptions`
```

Comment:
A lot of consumers don't use `noImplicitOverride`. I think we'll still want the rule to be able to work for them... which means it'll have to use [rule type checking](https://typescript-eslint.io/custom-rules#type-checking) when there isn't an `override` operator on a method.

Roughly, that means:

1. Grabbing the type of the extended class
2. Checking whether it has a method / function property under the same name

Reply:
Let's keep conversations threaded, please, so it's easier to have multiple at a time & mark them as open or resolved.

As to _how_ to grab the type of the extended class: you'll need to use the type checker. Have you read our resources on doing that?
* https://typescript-eslint.io/blog/asts-and-typescript-eslint
* https://sourcegraph.com/notebooks/Tm90ZWJvb2s6MTA2OA==
* https://typescript-eslint.io/custom-rules#typed-rules